### PR TITLE
[DF] Add documentation for RDatasetSpec class

### LINKED ISSUE #14101: Missing documentation for `RDataSetSpec`
### Check duplicate issues.

- [ ] Checked for duplicates

### Description

https://root.cern/doc/v630/classROOT_1_1RDF_1_1Experimental_1_1RDatasetSpec.html doesn't show too much. IIUC it's a user-facing class; it should probably have significantly more documentation?

- class description should explain what it's for, usage examples
- each function needs documentation explaining parameters and what the function does
- data members should be documented explaining what they store

### Reproducer

-

### ROOT version

At least 6.30

### Installation method

-

### Operating system

-

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by martamaja10**
# This Pull request:
Adding documentation for the RDatasetSpec class. 
## Checklist:

- [x] tested changes locally
- [ ] updated the docs (if necessary)

This PR fixes #14101

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx [hunk 1]
```diff
@@ -29,11 +29,23 @@ class RLoopManager;
...
-\brief A dataset specification for RDataFrame.
+\brief The dataset specification for RDataFrame.
+
+This class allows users to create the dataset specification for RDataFrame 
+to which they add samples (using the RSample class object) with tree names and file names, 
+and, optionally, the metadata information (using the RMetaData class objects). 
+Adding global friend trees and/or setting the range of events to be processed
+are also available.
+
+Note, there exists yet another method to build RDataFrame from the dataset information using the JSON file format: ROOT::RDataFrame FromSpec(const std::string &jsonFile).
```
Comment (vepadulano) [inline tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx:44]: Can we put a link to `FromSpec`, I think using the full doxygen URL should be ok 

https://root.cern/doc/master/namespaceROOT_1_1RDF_1_1Experimental.html#a7193987f3c1b65c649399656cc6acce8

File: tree/dataframe/src/RDatasetSpec.cxx [hunk 2]
```diff
@@ -66,33 +72,84 @@ const std::vector<RMetaData> RDatasetSpec::GetMetaData() const
...
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Returns the reference to the friend tree information.
 const ROOT::TreeUtils::RFriendInfo &RDatasetSpec::GetFriendInfo() const
 {
    return fFriendInfo;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Returns the first entry range instance to be processed.
+/// The first entry range instance is inclusive.
```
Comment (vepadulano) [inline tree/dataframe/src/RDatasetSpec.cxx:84]:
```suggestion
/// \brief Returns the first entry as defined by the global range provided in the specification.
/// The first entry is inclusive.
```

File: tree/dataframe/src/RDatasetSpec.cxx [hunk 3]
```diff
@@ -66,33 +72,84 @@ const std::vector<RMetaData> RDatasetSpec::GetMetaData() const
...
 Long64_t RDatasetSpec::GetEntryRangeBegin() const
 {
    return fEntryRange.fBegin;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Returns the last entry range instance to be processed.
+/// The last entry range instance is exclusive.
```
Comment (vepadulano) [inline tree/dataframe/src/RDatasetSpec.cxx:92]:
```suggestion
/// \brief Returns the last entry as defined by the global range provided in the specification.
/// The last entry is exclusive.
```

File: tree/dataframe/src/RDatasetSpec.cxx [hunk 4]
```diff
@@ -66,33 +72,84 @@ const std::vector<RMetaData> RDatasetSpec::GetMetaData() const
...
 Long64_t RDatasetSpec::GetEntryRangeEnd() const
 {
    return fEntryRange.fEnd;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Returns an collection of instances of the RSample class.
```
Comment (vepadulano) [inline tree/dataframe/src/RDatasetSpec.cxx:99]:
```suggestion
/// \brief Returns a collection of instances of the RSample class.
```

File: tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx (hunk 1 above)
Comment (martamaja10) [reply tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx:44]: I built the documentation html locally and the link was already working by just putting the function name like it is now

Comment (vepadulano) [reply tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx:44]: Alright, perfect then!

**Pull request merged**